scripts/Diffraction/isis_powder/polaris.py

```diff
@@ -0,0 +1,373 @@
+        return normalised_ws
+
+    def apply_solid_angle_efficiency_corr(self, ws_to_correct, run_details):
+        if not self._apply_solid_angle:
+            return ws_to_correct
+
+        if not run_details or not os.path.isfile(run_details.solid_angle_corr):
+            corrections = self.generate_solid_angle_corrections(run_details)
+        else:
+            corrections = mantid.Load(Filename=run_details.solid_angle_corr)
+
+        corrected_ws = mantid.Divide(LHSWorkspace=ws_to_correct, RHSWorkspace=corrections)
+        common.remove_intermediate_workspace(corrections)
+        common.remove_intermediate_workspace(ws_to_correct)
+        ws_to_correct = corrected_ws
+        return ws_to_correct
+
+    def generate_solid_angle_corrections(self, run_details):
+        solid_angle_vanadium_ws = common.load_current_normalised_ws(run_number_string=run_details.vanadium,
+                                                                    instrument=self)
+
+        corrections = _calculate_solid_angle_efficiency_corrections(solid_angle_vanadium_ws)
+        mantid.SaveNexusProcessed(InputWorkspace=corrections, Filename=run_details.solid_angle_corr)
+
+        common.remove_intermediate_workspace(solid_angle_vanadium_ws)
+        return corrections
+
+    def correct_sample_vanadium(self, focused_ws, index, vanadium_ws=None):
+        spectra_name = "sample_ws-" + str(index + 1)
+        mantid.CropWorkspace(InputWorkspace=focused_ws, OutputWorkspace=spectra_name,
+                             StartWorkspaceIndex=index, EndWorkspaceIndex=index)
+
+        if vanadium_ws:
+            van_rebinned = mantid.RebinToWorkspace(WorkspaceToRebin=vanadium_ws, WorkspaceToMatch=spectra_name)
+            mantid.Divide(LHSWorkspace=spectra_name, RHSWorkspace=van_rebinned, OutputWorkspace=spectra_name)
+            common.remove_intermediate_workspace(van_rebinned)
+
+        return spectra_name
+
+    def _spline_background(self, focused_vanadium_ws, spline_number, instrument_version=''):
+
+        if spline_number is None:
+            spline_number = 100
+
+        mode = "spline"  # TODO support spline modes for all instruments
+        extracted_spectra = _extract_bank_spectra(focused_vanadium_ws, self._number_of_banks)
+
+        if mode == "spline":
+            output = self._mask_spline_vanadium_ws(vanadium_spectra_list=extracted_spectra,
+                                                   spline_coefficient=spline_number)
+        else:
+            raise NotImplementedError("Other vanadium processing methods not yet implemented")
+
+        for ws in extracted_spectra:
+            common.remove_intermediate_workspace(ws)
+
+        return output
+
+    def _generate_vanadium_absorb_corrections(self, calibration_full_paths, ws_to_match):
+        absorb_ws = mantid.CloneWorkspace(InputWorkspace=ws_to_match)
+
+        # TODO move all of this into defaults
+        cylinder_sample_height = str(4)
```

Comment:
👍 I haven't tried passing them as ints/floating points yet. Originally I was matching the original output of the Crystallography scripts and wanted to avoid the chance of floating point errors so opted to pass them by string instead. 
Additionally these were going to be moved out into an advanced configuration file with the future work so are only here as a temporary stopgap so I'll remove the casts in the process too.